Fails to build with Go 1.25

👋  When trying to build it with Go 1.25 (rc3) in
* https://github.com/Homebrew/homebrew-core/pull/226636

The build [failed](https://github.com/Homebrew/homebrew-core/actions/runs/16788155185/job/47549802968?pr=226636#step:3:18792) with:
```
  # github.com/bytedance/sonic/internal/rt
  /Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/bytedance/sonic@v1.13.3/internal/rt/stubs.go:33:22: undefined: GoMapIterator
  /Users/brew/Library/Caches/Homebrew/go_mod_cache/pkg/mod/github.com/bytedance/sonic@v1.13.3/internal/rt/stubs.go:36:54: undefined: GoMapIterator
```

Mcphost is still using
https://github.com/mark3labs/mcphost/blob/d3f9759eb4cacf8da2cd4d0526414630ad146655/go.mod#L10

This should be upgraded to [1.14.0](https://github.com/bytedance/sonic/releases/tag/v1.14.0) (or later) to get upstream fix:
* https://github.com/bytedance/sonic/pull/808

Please note, that Go 1.25 is scheduled for release in August.

--- a/go.mod
+++ b/go.mod
@@ -7,23 +7,23 @@ toolchain go1.23.9
 require (
 	github.com/JohannesKaufmann/html-to-markdown v1.6.0
 	github.com/PuerkitoBio/goquery v1.10.3
-	github.com/bytedance/sonic v1.13.3
+	github.com/bytedance/sonic v1.14.0
 	github.com/charmbracelet/lipgloss v1.1.1-0.20250404203927-76690c660834
 	github.com/cloudwego/eino v0.3.41
 	github.com/cloudwego/eino-ext/components/model/claude v0.0.0-20250609074000-b7f307dffa18
 	github.com/cloudwego/eino-ext/components/model/ollama v0.0.0-20250609074000-b7f307dffa18
 	github.com/cloudwego/eino-ext/components/model/openai v0.0.0-20250609074000-b7f307dffa18
+	github.com/getkin/kin-openapi v0.118.0
 	github.com/mark3labs/mcp-filesystem-server v0.11.1
 	github.com/mark3labs/mcp-go v0.34.0
 	github.com/ollama/ollama v0.5.12
 	github.com/spf13/cobra v1.8.1
 	github.com/spf13/viper v1.20.1
+	github.com/tidwall/gjson v1.18.0
 	golang.org/x/term v0.31.0
 	google.golang.org/genai v1.10.0
-	gopkg.in/yaml.v3 v3.0.1 // indirect
+	gopkg.in/yaml.v3 v3.0.1
 )
-
-require github.com/getkin/kin-openapi v0.118.0
 
 require (
 	cloud.google.com/go v0.116.0 // indirect
@@ -48,7 +48,7 @@ require (
 	github.com/aws/aws-sdk-go-v2/service/sts v1.33.9 // indirect
 	github.com/aws/smithy-go v1.22.1 // indirect
 	github.com/aymerick/douceur v0.2.0 // indirect
-	github.com/bytedance/sonic/loader v0.2.4 // indirect
+	github.com/bytedance/sonic/loader v0.3.0 // indirect
 	github.com/charmbracelet/colorprofile v0.2.3-0.20250311203215-f60798e515dc // indirect
 	github.com/charmbracelet/harmonica v0.2.0 // indirect
 	github.com/charmbracelet/x/cellbuf v0.0.13 // indirect
@@ -98,7 +98,6 @@ require (
 	github.com/spf13/afero v1.12.0 // indirect
 	github.com/spf13/cast v1.7.1 // indirect
 	github.com/subosito/gotenv v1.6.0 // indirect
-	github.com/tidwall/gjson v1.18.0 // indirect
 	github.com/tidwall/match v1.1.1 // indirect
 	github.com/tidwall/pretty v1.2.1 // indirect
 	github.com/tidwall/sjson v1.2.5 // indirect

